platform/platform-migratable/src/test/java/org/codice/ddf/platform/migratable/impl/PlatformMigratableTest.java

```diff
@@ -0,0 +1,273 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMetadata;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.migration.util.MigratableUtil;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Matchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PlatformMigratableTest {
+
+    private static final String DDF_BASE_DIR = "ddf";
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final Path WS_SECURITY_DIR_REL_PATH = Paths.get("etc", "ws-security");
+
+    private static final Path SYSTEM_PROPERTIES_REL_PATH = Paths.get("etc", "system.properties");
+
+    private static final Path USERS_PROPERTIES_REL_PATH = Paths.get("etc", "users.properties");
+
+    private static final Path EXPORTED_DIR_REL_PATH = Paths.get("etc", "exported");
+
+    private static final String DESCRIPTION = "Exports system files";
+
+    private static final boolean IS_OPTIONAL = false;
+
+    private Path ddfHome;
+
+    private Path exportDirectory;
+
+    @Before
+    public void setup() throws Exception {
+        ddfHome = Paths.get(DDF_BASE_DIR);
+        exportDirectory = ddfHome.resolve(EXPORTED_DIR_REL_PATH);
+    }
+
+    @Test
+    public void testExportValidRelativePaths() throws Exception {
+        // Setup
+        MigratableUtil mockMigratableUtil = mock(MigratableUtil.class);
+        PlatformMigratable platformMigratable = new PlatformMigratable(DESCRIPTION, IS_OPTIONAL,
+                mockMigratableUtil);
+
+        // Perform Test
+        platformMigratable.export(exportDirectory);
+
+        // Verify ws-security directory is exported
+        assertWsSecurityExport(mockMigratableUtil);
+
+        // Verify system.properties and users.properties are exported
+        assertSystemPropertiesFilesExport(mockMigratableUtil);
+
+        // Verify keystore and truststore are exported
+        assertKeystoresExport(mockMigratableUtil);
+    }
+
+    /**
+     * Verify that if an absolute path is encountered during the export, a warning is returned.
+     */
+    @Test
+    public void testExportWarningsReturnedWhenExportingKeystoreAndTrustsore() throws Exception {
+        // Setup
+        MigratableUtil mockMigratableUtil = mock(MigratableUtil.class);
+        MigrationWarning expectedMigrationWarning1 = new MigrationWarning("warning 1");
+        doAnswer(new Answer<Void>() {
```

Comment:
Should we create an inner class for this that takes in the `expectedWarning` as a parameter and implements the `answer()` method? We should be able use it almost everywhere, shrink the test class quite a bit and make the code more readable.